### PR TITLE
[Notebook] V2.0 development #2666

### PR DESCRIPTION
**Pull request by nikhilmandlik**
**Original Issue:** https://github.com/nasa/openmct/issues/2666

### Known Bugs:
* [x]  **Focus on newly added entry** 
  - _**Issue:**_ https://github.com/nasa/openmct/issues/2819
  - **_PR_:** https://github.com/nasa/openmct/pull/2824
* [ ]  **Remove 'new notebook entry' button inside Preview.**
  - **_Issue_:** https://github.com/nasa/openmct/issues/2821
  - **_PR_:** N/A
* [ ]  **Entries filter**
  - **_Issue_:** https://github.com/nasa/openmct/issues/2820
  - **_PR_:** N/A
* [ ]  **Remove default section/page from localstorage on notebook delete** 
  - **_Issue_:** https://github.com/nasa/openmct/issues/2822
  - **_PR_:** N/A

### CODE REVIEW

**Comment by akhenry** [comment]
@nikhilmandlik Needs a merge from master. Also looks like there are some code style issues.

File: src/plugins/notebook/components/notebook-menu-switcher.vue [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    },
+    methods: {
+        async setNotebookTypes() {
+            const notebookTypes = [];
+            let defaultPath = '';
+            const defaultNotebook = getDefaultNotebook();
+
+            if (defaultNotebook) {
+                const domainObject = await this.openmct.objects.get(defaultNotebook.notebookMeta.identifier)
+                    .then(d => d);
```
Comment (akhenry) [inline src/plugins/notebook/components/notebook-menu-switcher.vue]: Why does this `then` statement exist?

File: src/ui/router/Browse.js [hunk 2]
```diff
@@ -9,22 +9,31 @@ define([
...
+            routingInProgress = true;
             let navigatePath = results[1];
             navigateToPath(navigatePath, params.view);
+            navigateToObject(null, null, params);
         });
 
-        openmct.router.on('change:params', function (newParams, oldParams, changed) {
+        openmct.router.on('change:params', navigateToObject);
+
+        function navigateToObject(newParams, oldParams, changed) {
```
Comment (akhenry) [inline src/ui/router/Browse.js]: I don't think these changes are needed any more ?

File: src/ui/utils/tree.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    for (const composition of domainObject.composition) {
+        hasObject = hasObject
+            || await openmct.objects.get(composition)
+                .then(object => getObject(openmct, object, identifier));
+    }
+
+    return hasObject;
+}
+
+export const searchObjectInTree = async (openmct, identifier) => {
```
Comment (akhenry) [inline src/ui/utils/tree.js]: Is this still needed? I don't think we need to do an exhaustive tree search any more?

File: src/ui/router/Browse.js (hunk 2 above)
Comment (nikhilmandlik) [reply src/ui/router/Browse.js]: Tested: Still need this change for proper routing.

File: src/ui/utils/tree.js (hunk 3 above)
Comment (nikhilmandlik) [reply src/ui/utils/tree.js]: removed.

File: src/ui/router/Browse.js (hunk 2 above)
Comment (nikhilmandlik) [reply src/ui/router/Browse.js]: updated function name, as per discussion.

**Comment by akhenry** [comment]
## Reviewer Checklist

* Changes appear to address issue? Y
* Appropriate unit tests included? N - followup filed https://github.com/nasa/openmct/issues/2825
* Code style and in-line documentation are appropriate? Y
* Commit messages meet standards? Y

**Pull request merged**